### PR TITLE
Append azure error to query unsuccessful message

### LINKED ISSUE #269: Expired credentials give ugly [object Object] error in query builder
**What happened**:

When trying to make a new query using a data source that has expired credentials, the query editor errors out with `Could not load datasource schema: [object Object]`

**What you expected to happen**:

It displays an informative error message!

**How to reproduce it (as minimally and precisely as possible)**:

1. Configure a data source using expired credentials. (Note for Grafanistas: currently our "Azure Data Explorer Github" credentials have expired
2. Create a new panel using that data source

**Screenshots**

image

The initial request from the data source appears to fail in the frontend with error 401:

```json
{
    "Message": "",
    "Error": "HTTP error: 401 Unauthorized - {\"Message\":\"Authorization has been denied for this request.\"}",
    "StatusCode": 401
}
```

### PR DESCRIPTION
**Pull request by iwysiu**
Fixes #269

### CODE REVIEW

**Review by sunker — COMMENTED** [review]
Looks good! I have not tested this manually, but I guess you have? Should also be possible to test this in resource_handler_test.go?

File: pkg/azuredx/resource_handler.go [hunk 1]
```diff
@@ -62,6 +63,9 @@ func (adx *AzureDataExplorer) getDatabases(rw http.ResponseWriter, req *http.Req
 }
 
 func respondWithError(rw http.ResponseWriter, code int, message string, err error) {
+	if err != nil {
+		message = fmt.Sprintf("%s: %s", message, err)
+	}
 	httpError := models.NewHttpError(message, code, err)
```
[the remark this answers is not in the document]
Comment (iwysiu) [reply pkg/azuredx/resource_handler.go:69]: The error gets put in its own error field on the returned json, but the only field displayed to users is the message field. I've edited the test to also check the message field

**Review by iwysiu — COMMENTED** [review]
Test added!

**Review by andresmgot — APPROVED** [review]
LGTM!

**Comment by andresmgot** [comment]
BTW, to fix the issue in the build, it's because the environment is now running with node 16. You need to upgrade (at least) the @grafana/toolkit dependency. See: https://github.com/grafana/grafana/issues/40764#issuecomment-984554896

File: pkg/azuredx/resource_handler.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (shirleyleu) [reply pkg/azuredx/resource_handler.go:69]: Thanks for the clarification!

I am not sure, but I get the feeling that the `httpError` should be constructed in only one place -- either:
1. in this function `respondWithError` and remove `NewHttpError` function completely
2. or the err nil check happens in `NewHttpError` and that's where the error message is appended

I don't mean to nitpick, just food for thought. Happy to discuss further separately.

**Review by andresmgot — APPROVED** [review]

**Comment by andresmgot** [comment]
and now there is a lot of lint issues :sweat_smile: those are easy to fix but you it's probably better to do that in a different PR to not pollute your original change here (and the rebase here). Let me know if you want to pair for this one!

**Pull request merged**